types/coin.go

```diff
@@ -639,29 +638,30 @@ func mustValidateDenom(denom string) {
 	}
 }
 
-// ParseCoin parses a cli input for one coin type, returning errors if invalid or on an empty string
+// ParseCoinNormalized parses and normalize a cli input for one coin type, returning errors if invalid or on an empty string
 // as well.
 // Expected format: "{amount}{denomination}"
-func ParseCoin(coinStr string) (coin Coin, err error) {
+func ParseCoinNormalized(coinStr string) (coin Coin, err error) {
```

Comment:
Why are we using decimal types in this function?

Reply:
We already support decimal coins in `ParseCoinsNormalized`, so it's just keep it consistent with that.
Parse decimal coins so we can write `0.1atom` in place of `10...0utom`.

Reply:
I mean why are we using decimal coins in a function that returns `Coin`? This seems like a codesmell to me. Can we not have separate functions for `Coin` and `DecCoin`?

Reply:
It's a higher level function which involves both `DecCoin` and `Coin`, the behavior is clear:
- parse dec coin
- do conversion
- truncate and take the int part
mainly used to parse user inputs in cli, I guess it's just a matter of where to put this code?

Reply:
Yes, we should have distinct separate functions for these.

Reply:
> Yes, we should have distinct separate functions for these.

how about `types/utils.go` or `client/utils.go`?

Reply:
> Yes, we should have distinct separate functions for these.

I think @alexanderbez meant separate functions for each of the `DecCoin` and `Coin` types?

Reply:
> > Yes, we should have distinct separate functions for these.
> 
> I think @alexanderbez meant separate functions for each of the `DecCoin` and `Coin` types?

oh, you are right, most of it is just duplicate of `ParseDecCoin`, changed.

Reply:
Yes, just two separate methods :)

Reply:
> Yes, just two separate methods :)

Err, sorry, what do you mean by that? currently what I want is parse decimal coins and convert it to int coins, I think it's best for cli UX, do you mean we should keep the basic version of `ParseCoin` and `ParseDecCoin`?

Reply:
Yes. Where in the code do we convert between the two often? 

There should be method to parse/normalize `Coin(s)` and a separate method to parse/normalize `DecCoin(s)`.

Reply:
> Yes. Where in the code do we convert between the two often?

I think it's only used in cli to process user input.

Reply:
I see. So all user input-based coins (e.g. `tx send`) will have the coin amount (e.g. `1.2atom`, `4000uatom`, etc...) will have to go through `ParseCoinNormalized` and only that function? If so, then I think this is OK.

Reply:
> I see. So all user input-based coins (e.g. `tx send`) will have the coin amount (e.g. `1.2atom`, `4000uatom`, etc...) will have to go through `ParseCoinNormalized` and only that function? If so, then I think this is OK.

yes, `ParseCoinNormalized` and `ParseCoinsNormalized`.